src/qt/walletmodel.cpp

```diff
@@ -756,3 +779,296 @@ bool WalletModel::getDefaultWalletRbf() const
 {
     return fWalletRbf;
 }
+
+bool WalletModel::nameAvailable(const QString &name)
+{
+    JSONRPCRequest jsonRequest;
+    UniValue params (UniValue::VOBJ);
+    UniValue res, array, isExpired;
+
+    const std::string strName = name.toStdString();
+    params.push_back (Pair("name", strName));
+    jsonRequest.strMethod = "name_show";
+    jsonRequest.params = params;
+    jsonRequest.fHelp = false;
+
+    try {
+        res = tableRPC.execute(jsonRequest);
+
+    } catch (const UniValue& e) {
+        return true;
+    }
+
+    isExpired = find_value( res, "expired");
+    if(isExpired.get_bool())
+      return true;
+
+    return false;
+}
+
+NameNewReturn WalletModel::nameNew(const QString &name)
+{
+    std::string strName = name.toStdString ();
+    std::string data;
+
+    JSONRPCRequest jsonRequest;
+    UniValue params(UniValue::VOBJ);
+    std::vector<UniValue> values;
+    UniValue res, txid, rand;
+
+    NameNewReturn retval;
+
+    params.push_back (Pair("name", strName));
+
+    jsonRequest.strMethod = std::string("name_new");
+    jsonRequest.params = params;
+    jsonRequest.fHelp = false;
+
+    try {
+        res = name_new (jsonRequest);
+    } catch (const UniValue& e) {
+        UniValue message = find_value( e, "message");
+        std::string errorStr = message.get_str();
+        LogPrintf ("nameNew error: %s\n", errorStr.c_str());
+        retval.err_msg = errorStr;
+        retval.ok = false;
+        return retval;
+    }
+
+    retval.ok = true;
+
+    values = res.getValues ();
+    txid = values[0];
+    rand = values[1];
+
+    // txid
+    retval.hex = txid.get_str ();
+    // rand
+    retval.rand = rand.get_str ();
+
+    return retval;
+}
+
+// this gets called from configure names dialog after name_new succeeds
+QString WalletModel::nameFirstUpdatePrepare(const QString& name, const QString& data)
+{
+    if (data.isEmpty())
+        return tr("Data cannot be blank.");
+
+    const std::string strName = name.toStdString ();
+    const std::string strData = data.toStdString();
+    NameNewReturn updatedNameNewReturn;
+
+    // we need to write out everything in NameNewReturn to the wallet, essentially
+    // that's where wallet->WriteNameFirstUpdate comes in
+    // it gets loaded at wallet start, and as long as we keep it updated as
+    // things happen, we should be able to rely on it
+
+    LOCK(cs_main);
+    // pendingNameFirstUpdate is set by the time we get here inside managenamespage.cpp:134
```

Comment:
I think you shouldn't refer to the line number in this comment, as that will get outdated very quickly.  Can you please reformulate it in a way that makes it clear what place in the code you mean without a line number -- e. g., function name / description of the operation there?